lib/core/test/unit/Cardano/WalletSpec.hs

```diff
@@ -489,6 +497,68 @@ walletListTransactionsSorted wallet@(wid, _, _) _order (_mstart, _mend) history
                 (\(tx, meta) -> (txId tx, slotIdTime (meta ^. #slotId))) <$> history
         times `shouldBe` expTimes
 
+
+{-------------------------------------------------------------------------------
+                        Properties of tx fee estimation
+-------------------------------------------------------------------------------}
+
+-- | Properties of 'estimateFeeForCoinSelection':
+-- 1. There is no coin selection with a fee above the estimated maximum.
+-- 2. The minimum estimated fee is no greater than the maximum estimated fee.
+-- 3. Around 10% of fees are below the estimated minimum.
+prop_estimateFee :: NonEmptyList (Either String FeeGen) -> Property
+prop_estimateFee (NonEmpty results) = case actual of
+    Left err -> label "errors: all" $
+        Left err === head results
+    Right estimation@(W.FeeEstimation minFee maxFee) ->
+        label ("errors: " <> if any isLeft results then "some" else "none") $
+        counterexample (show estimation) $
+            maxFee <= maximum (map (getRight 0) results) .&&.
```

Comment:
I could be wrong here, but shouldn't this be an equality test?

So:
```suggestion
            maxFee == maximum (map (getRight 0) results) .&&.
```

Reply:
It can't be an equality test because the real maximum fee might not have been sampled.

Reply:
> It can't be an equality test because the real maximum fee might not have been sampled.

Okay that makes sense. Thanks for clarifying!